### DISCUSSION TITLE
Don't receive cookie from custom route

### DISCUSSION THREAD

**Discussion by starredev** [pocketbase/pocketbase#3850] [Q&A] [answered]
Hello, I just created my own TOTP authentication plugin in golang together with pocketbase api (this was my first experience with go). Normally when I authenticate myself through authWithPassword js-sdk the cookie is automaticly set when I receive the response. 

In this case i've created a custom route but in my vue project i'm not getting the cookie, could you help me?

````
        e.Router.POST("/auth-login-totp", func(c echo.Context) error {
    
            data := &struct {
				Email    string `form:"email" json:"email"`
				Password string `form:"password" json:"password"`
                TOTPCode string `form:"totpCode" json:"totpCode"`
			}{}

			// read the request data
			if err := c.Bind(data); err != nil {
				return apis.NewBadRequestError("Failed to read request data", err)
			}

            record, err := app.Dao().FindFirstRecordByData("users", "email", data.Email)
            if err != nil || !record.ValidatePassword(data.Password) {
                return apis.NewBadRequestError("Invalid credentials", err)
            }

            if data.TOTPCode == "" && record.Get("secret_otp") != ""  {
                return c.JSON(http.StatusOK, map[string]string{"message": "Authenticator enabled", "status": "authenticator_enabled"})
            }

            valid := totp.Validate(data.TOTPCode, record.Get("secret_otp").(string))
            if !valid {
                return apis.NewForbiddenError("Google authenticator code not correct", nil)
            }

            // generate a new auth token for the found user record
            token, err := tokens.NewRecordAuthToken(app, record)
            if err != nil {
                return apis.NewBadRequestError("Failed to create an auth token", err)
            }

            // set it as cookie
			cookie := &http.Cookie{
				Name:     "pb_auth", // rename with the name of your cookie
				Value:    token,
				Secure:   true,
				HttpOnly: true,
				SameSite: http.SameSiteStrictMode,
				// you can use the token duration or any other expiration (eg. only 1 day)
				Expires:  time.Now().Add(time.Duration(app.Settings().RecordAuthToken.Duration) * time.Second),
			}

			c.SetCookie(cookie)

			return apis.RecordAuthResponse(app, c, record, nil)
        }, apis.ActivityLogger(app))
````

**Comment by ganigeorgiev** [comment]
If the Vue app is accessible at different host+port address then most likely is caused by `http.SameSiteStrictMode`. Try to temp replace it with `http.SameSiteLaxMode`.

But why do you need a cookie in the first place?

**Comment by starredev** [comment]
> If the Vue app is accessible at different host+port address then most likely is caused by `http.SameSiteStrictMode`. Try to temp replace it with `http.SameSiteLaxMode`.
> 
> But why do you need a cookie in the first place?

I'm not sure how to handle this but when i authenticate myself through the collection('users').authWithPassword() I receive pocketbase_auth in my localStorage. With my custom totp authentication it don't store pocketbase_auth what means that when I refresh my page, the authentication is lost.

````const auth = await pb.send('/auth-login-totp', { method: "POST", body: formData })````

**Reply by ganigeorgiev** [reply, marked as answer]
You can persist the result in the authStore without a cookie by doing something like:

```js
const auth = await pb.send('/auth-login-totp', { method: "POST", body: formData })

pb.authStore.save(auth.token, auth.record)
```
